plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java

```diff
@@ -334,6 +338,10 @@ public static Object getField(DateTimeZone localZone, Type type, Block block, in
         if (type instanceof TimestampType timestampType) {
             return getHiveTimestamp(localZone, timestampType, block, position);
         }
+        if (type instanceof TimestampWithTimeZoneType) {
+            checkArgument(type.equals(TIMESTAMP_TZ_MILLIS));
```

Comment:
Agreed. We call here from `io.trino.plugin.deltalake.AbstractDeltaLakePageSink#createPartitionValues`.
For example i am not convinced this is the appropriate check for Delta
https://github.com/trinodb/trino/blob/431f3e066e87062953ae4529e88a3f4220adef41/plugin/trino-hive/src/main/java/io/trino/plugin/hive/util/HiveWriteUtils.java#L285-L289

in delta context the value gets used to
- make partition-like path
- write to addEntry json (right?) 

so the logic could indeed be Delta-specific.

AFAIAC, this can go follow-up

Reply:
I will send a follow-up PR.